util/imagetools/printers.go

```diff
@@ -31,53 +27,46 @@ type Printer struct {
 	name   string
 	format string
 
-	raw       []byte
-	ref       reference.Named
-	manifest  ocispecs.Descriptor
-	index     ocispecs.Index
-	platforms []ocispecs.Platform
+	insres   *result
+	raw      []byte
+	ref      reference.Named
+	manifest ocispecs.Descriptor
+	index    ocispecs.Index
 }
 
 func NewPrinter(ctx context.Context, opt Opt, name string, format string) (*Printer, error) {
 	resolver := New(opt)
 
-	ref, err := parseRef(name)
+	insres, err := newLoader(resolver.resolver()).Load(ctx, name)
 	if err != nil {
 		return nil, err
 	}
 
-	dt, manifest, err := resolver.Get(ctx, name)
+	ref, err := parseRef(name)
 	if err != nil {
 		return nil, err
 	}
 
-	var index ocispecs.Index
-	if err = json.Unmarshal(dt, &index); err != nil {
+	dt, mfst, err := resolver.Get(ctx, ref.String())
```

Comment:
As I read this, are we fetching the image index/manifest here as well as in `insres`? Not quite sure I understand what `insres` is for, the name doesn't indicate a lot.